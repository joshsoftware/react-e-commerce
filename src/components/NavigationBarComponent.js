import React, { useState } from 'react';
import NavbarBrandComponent from './NavbarBrandComponent';
import NavbarTogglerComponent from './NavbarTogglerComponent';
import NavComponent from './NavComponent';
import logo from '../images/logo.jpg';
import { Navbar } from 'reactstrap';
import PropTypes from 'prop-types';

const NavigationBarComponent = (props) => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  const { color, expand } = props;

  return (
    <Navbar color={color} expand={expand}>
      <NavbarBrandComponent url={'/'} logo={logo} />
      <NavbarTogglerComponent toggle={toggle} />
      <NavComponent />
    </Navbar>
  );
};

export default NavigationBarComponent;

NavigationBarComponent.propTypes = {
  color: PropTypes.string.isRequired,
  expand: PropTypes.string
};