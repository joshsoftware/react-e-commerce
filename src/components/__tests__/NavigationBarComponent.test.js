import NavigationBarComponent from '../NavigationBarComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('NavigationBar Component', () => {
  it('Must return Navigation bar', () => {
    const color = 'dark';
    const { asFragment } = render(<NavigationBarComponent color={color} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
