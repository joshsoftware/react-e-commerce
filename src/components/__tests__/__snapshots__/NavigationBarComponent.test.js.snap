// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Component Must return Navigation bar 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <img
        alt="img"
        class="rounded-circle"
        src="logo.jpg"
        style="width: 70px;"
      />
    </a>
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <ul
      class="mr-auto navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/components"
        >
          Shop
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
    <ul
      class="nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/about"
        >
          <img
            alt="img"
            class="rounded-circle"
            src="cart.jpg"
            style="width: 70px;"
          />
        </a>
      </li>
      <li
        class="dropdown nav-item"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle nav-link"
          href="#"
        >
          <img
            alt="img"
            class="rounded"
            src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
            style="width: 50px;"
          />
        </a>
        <div
          aria-hidden="true"
          class="dropdown-menu dropdown-menu-right"
          role="menu"
          tabindex="-1"
        >
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Profile
          </button>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Logout
          </button>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
        </div>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
